python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+        exposureTimes.append(0)
+        mjds.append((refEpoch if (refEpoch is not None) else medianEpoch))
+        observatories.append(np.array([0, 0, 0]))
+        identity = wcsfit.IdentityMap()
+        icrs = wcsfit.SphericalICRS()
+        refWcs = wcsfit.Wcs(identity, icrs, "Identity", np.pi / 180.)
+        wcss.append(refWcs)
+
+        extensionVisits.append(len(exposureNames) - 1)
+        extensionDetectors.append(-1)  # REFERENCE device must be -1
+        visits.append(-1)
+        detectors.append(-1)
+        extensionType.append('REFERENCE')
+
+        # Make a table of information to use elsewhere in the class
+        self.extensionInfo = pd.DataFrame({'visit': visits, 'detector': detectors,
+                                           'visitIndex': extensionVisits,
+                                           'detectorIndex': extensionDetectors,
+                                           'wcs': wcss, 'extensionType': extensionType})
+
+        # Make the exposureHelper object to use in the fitting routines
+        exposuresHelper = wcsfit.ExposuresHelper(exposureNames,
+                                                 fieldNumbers,
+                                                 instrumentNumbers,
+                                                 self.ras,
+                                                 self.decs,
+                                                 airmasses,
+                                                 exposureTimes,
+                                                 mjds,
+                                                 observatories)
+
+        return exposuresHelper, medianEpoch
+
+    def _load_refCat(self, associations, center, radius, epoch=None, fieldIndex=0, instrumentIndex=-1):
+        """Load the reference catalog and add reference objects to the
+        `wcsfit.FoFClass` object.
+
+        Parameters
+        ----------
+        associations : `wcsfit.FoFClass`
+            Object to which to add the catalog of reference objects
+        center : `lsst.geom.SpherePoint`
+            Center of the circle in which to load reference objects
+        radius : `lsst.sphgeom._sphgeom.Angle`
+            Radius of the circle in which to load reference objects
+        epoch : `float`
+            MJD to which to correct the object positions
+        """
+        formattedEpoch = astropy.time.Time(epoch, format='mjd')
+
+        refFilter = self.config.anyFilterMapsToThis
+        skyCircle = self.refObjectLoader.loadSkyCircle(center, radius, refFilter, epoch=formattedEpoch)
+
+        selected = self.referenceSelector.run(skyCircle.refCat)
+        # Need memory contiguity to get reference filters as a vector.
+        if not selected.sourceCat.isContiguous():
+            refCat = selected.sourceCat.copy(deep=True)
+        else:
+            refCat = selected.sourceCat
+
+        if self.config.useProperMotion:
+            # GBDES cannot currently accommodate catalogs with a mixture of PM
```

Comment:
Is there a ticket for this?  Or is this okay because all the gaia stars have a PM measurement?  Also, is this a limitation on all the stars, or just the refstars?

Reply:
I've actually been experimenting with this, and I think that `gbdes` can handle a mix of pm and non-pm sources. I asked Gary about it, but haven't heard back yet. I think the way Gary was running he only used Gaia sources with proper motion, but you do lose a lot (about 40% in the footprint of the test data I've been using). I'll make a ticket depending on what he says.

Reply:
That is an interesting point.  And I would say that this should not be True for HSC, the epoch from field to field varies by up to several years, and this does show up in the Gaia residuals if you're not careful.   (I've been ascribing this to PM; maybe wrong...?). But the HSC observations are not continuous, and the mean epoch of any field can be quite different from Gaia reference epoch, so the same argument holds.

Reply:
Ok, I'm going to set it to exclude non-PM objects for now, though I think it would be worth doing a small study in the future to see what the bias vs. improvement in fit tradeoff is for HSC.